posthog/api/cohort.py

```diff
@@ -283,13 +286,45 @@ def persons(self, request: Request, **kwargs) -> Response:
         elif not filter.limit:
             filter = filter.shallow_clone({LIMIT: 100})
 
-        query, params = PersonQuery(filter, team.pk, cohort=cohort).get_query(paginate=True)
+        if posthoganalytics.feature_enabled(
+            "load-person-fields-from-clickhouse",
+            None if request.user.is_anonymous else request.user.distinct_id,
+            person_properties={"email": request.user.email},
+        ):
+            person_query = PersonQuery(
+                filter,
+                team.pk,
+                extra_fields=[
+                    "created_at",
+                    "properties",
+                    "is_identified",
+                ],
+                include_distinct_ids=True,
+            )
+            paginated_query, paginated_params = person_query.get_query(paginate=True, filter_future_persons=True)
+            serialized_actors = insight_sync_execute(
+                paginated_query,
+                {**paginated_params, **filter.hogql_context.values},
+                filter=filter,
+                query_type="cohort_persons",
+                team_id=team.pk,
+            )
+            persons = []
+            for p in serialized_actors:
+                person = Person(uuid=p[0], created_at=p[1], is_identified=p[2], properties=json.loads(p[3]))
+                person._distinct_ids = p[4]  # type: ignore
+                persons.append(person)
+
+            serialized_actors = serialize_people(team, data=persons)
+            _should_paginate = len(serialized_actors) >= filter.limit
+        else:
+            query, params = PersonQuery(filter, team.pk, cohort=cohort).get_query(paginate=True)
```

Comment:
hmm, as far as I can tell, nothing is changing with this query, it's just the serialization and hydration of remaining fields afterwards, so can we pull this out of the feature flag check? same for person endpoint below

Reply:
No the query selects all fields here? I don't think we need to optimize for DRY here, as we'll remove all the duplicate stuff when removing the FF.